internal/core/v1beta1/distribution_service.go

```diff
@@ -66,20 +67,32 @@ func (d *DistributionService) CreateInvalidation(ctx context.Context, distributi
 		return nil, err
 	}
 
-	// prepend prefix to all the paths
-	absolutePaths := make([]string, 0, len(paths))
+	cleanedPaths := make([]string, 0, len(paths))
+	invalidPaths := make([]string, 0)
+
 	for _, p := range paths {
-		// prevent users from doing funny business by going up a directory
-		if strings.Contains(p, "../") {
-			return nil, NewInvalidationError(BadRequestErrorCode, fmt.Errorf("invalid path"), errors.New("path cannot contain ../"))
+		// URL decode the path, and clean it
+		decodedPath, err := url.PathUnescape(p)
+		if err != nil {
+			return nil, NewInvalidationError(BadRequestErrorCode, errors.New("invalid encoded path"), fmt.Errorf("invalid encoded path: %v", p))
```

Comment:
```suggestion
			return nil, NewInvalidationError(BadRequestErrorCode, errors.New("invalid encoded path"), fmt.Sprintf("invalid encoded path: %v", p))
```

The second argument was just meant to be additional args for the error message presented to the user.  It does not need to be of an "error" type.  This can get cleaned up with the errors in a later PR.